test/services/submissionService/SubmissionServiceTestJson.scala

```diff
@@ -0,0 +1,368 @@
+    """{
+      |   "supportingDocumentTypes":[
+      |      "OriginalC88",
+      |      "OriginalC2",
+      |      "AmendedSubstituteEntryWorksheet"
+      |   ],
+      |   "supportingDocuments":[
+      |      {
+      |         "fileName":"TestDocument.pdf",
+      |         "downloadUrl":"http://some/location",
+      |         "uploadTimestamp":"2021-05-14T20:15:13.807",
+      |         "checksum":"the file checksum",
+      |         "fileMimeType":"application/pdf"
+      |      }
+      |   ]
+      |}""".stripMargin
+
+  val singleUnderpaymentDefermentSupportingDocumentationJson =
+    """{
+      |   "supportingDocumentTypes":[
+      |      "OriginalC88",
+      |      "OriginalC2",
+      |      "AmendedSubstituteEntryWorksheet",
+      |      "DefermentAuthorisation"
+      |   ],
+      |   "supportingDocuments":[
+      |      {
+      |         "fileName":"TestDocument.pdf",
+      |         "downloadUrl":"http://some/location",
+      |         "uploadTimestamp":"2021-05-14T20:15:13.807",
+      |         "checksum":"the file checksum",
+      |         "fileMimeType":"application/pdf"
+      |      },
+      |      {
+      |         "fileName":"TestDocument.pdf",
+      |         "downloadUrl":"http://some/location",
+      |         "uploadTimestamp":"2021-05-14T20:15:13.807",
+      |         "checksum":"the file checksum",
+      |         "fileMimeType":"application/pdf"
+      |      }
+      |   ]
+      |}""".stripMargin
+
+  val noSplitDefermentSupportingDocumentationJson =
+    """{
+      |   "supportingDocumentTypes":[
+      |      "OriginalC88",
+      |      "OriginalC2",
+      |      "AmendedSubstituteEntryWorksheet",
+      |      "AmendedC88",
+      |      "AmendedC2",
+      |      "InvoiceAirwayBillPreferenceCertificate",
+      |      "InvoiceAirwayBillPreferenceCertificate",
+      |      "Other",
+      |      "DefermentAuthorisation"
+      |   ],
+      |   "supportingDocuments":[
+      |      {
+      |         "fileName":"TestDocument.pdf",
+      |         "downloadUrl":"http://some/location",
+      |         "uploadTimestamp":"2021-05-14T20:15:13.807",
+      |         "checksum":"the file checksum",
+      |         "fileMimeType":"application/pdf"
+      |      },
+      |      {
+      |         "fileName":"TestDocument.pdf",
+      |         "downloadUrl":"http://some/location",
+      |         "uploadTimestamp":"2021-05-14T20:15:13.807",
+      |         "checksum":"the file checksum",
+      |         "fileMimeType":"application/pdf"
+      |      }
+      |   ]
+      |}""".stripMargin
+
+  val importerDefermentDetailsJson =
+    """{
+      |   "defermentType":"D",
+      |   "defermentAccountNumber":"D1234567"
+      |}""".stripMargin
+
+  val representativeNoSplitDefermentDetailsJson =
+    """{
+      |   "defermentType":"B",
+      |   "defermentAccountNumber":"B1234567"
+      |}""".stripMargin
+
+  val importerDeclarantDetailsJson =
```

Comment:
The naming was to distinguish it would be the declarant details for the importer. The representative portion of the code is covered mainly in the completeSubmission. Do you think I should add further tests to explicitly test representative in the sub sections as well?

Reply:
Importer part removed